libs/libc/stdio/lib_printf.c

```diff
@@ -59,11 +55,7 @@ int printf(FAR const IPTR char *fmt, ...)
   int     ret;
 
   va_start(ap, fmt);
-#if CONFIG_NFILE_STREAMS > 0
-  ret = vfprintf(stdout, fmt, ap);
-#else
-  ret = nx_vsyslog(LOG_INFO, fmt, &ap);
-#endif
+  ret = vdprintf(STDOUT_FILENO, fmt, ap);
```

Comment:
I already merged this, but it is not correct.  printf must used streams and buffere C I/O if available.

Reply:
Yes, you are right. I forget the I/O buffer.